sdk/src/cache/cache_client_builder.rs

```diff
@@ -58,17 +62,44 @@ impl CacheClientBuilder<NeedsCredentialProvider> {
 }
 
 impl CacheClientBuilder<ReadyToBuild> {
+    pub fn with_num_connections(self, num_connections: u32) -> CacheClientBuilder<ReadyToBuild> {
+        let grpc_configuration = self.0.configuration.transport_strategy.grpc_configuration;
+        let transport_strategy = TransportStrategy {
+            grpc_configuration: GrpcConfiguration {
+                num_channels: num_connections,
```

Comment:
Should this .clamp(1, 32) or something?

Reply:
should it? I can imagine some hard-core scenarios where someone might want like 100 connections?

Also, this kind of relates to the public-facing signature question that I raised above. If we really want to restrict this I would guess that a better UX would be to make that argument a `u8` or something. And if we wanted to restrict it to < 255 then I would expect us to do some validation at construction time and then panic if they pass too high of a value in. clamping seems like a surprising behavior for an end user to experience.

Reply:
whatever the strategy, it seems worth helping users make a cromulent client rather than allowing clients with 0 channels, resulting in an index out of bound panic later on, or clients with unreasonably high connection counts. Someone can obviously bypass it, but probably the sdk on main should notice if someone tries to make a client with 4294967295 connections.

Reply:
fixed in: https://github.com/momentohq/client-sdk-rust/pull/393/commits/c90273feb5bc3b6124e95da54f24d15a44a0bed2